webapp/app/js/compare.js

```diff
@@ -0,0 +1,494 @@
+    var extra = "";
+    if (signatureProps.job_group_symbol === "T-e10s") {
+      extra = " e10s";
+    }
+    var testName = signatureProps.test;
+    var subtestSignatures;
+    if (testName === undefined) {
+      testName = "summary";
+      subtestSignatures = signatureProps.subtest_signatures;
+    }
+    var name = signatureProps.suite + " " + testName +
+      " " + optionCollectionMap[signatureProps.option_collection_hash] + extra;
+
+    //Only keep summary signatures, filter in/out e10s and pgo
+    if (name.indexOf('summary') <= 0) {
+        return {};
+    }
+    if (e10s && (name.indexOf('e10s') <= 0)) {
+        return {};
+    } else if (!e10s && (name.indexOf('e10s') > 0)) {
+        return {};
+    }
+
+    //TODO: pgo is linux/windows only- what about osx and android
+    if (pgo && (name.indexOf('pgo') <= 1)) {
+        return {};
+    } else if (!pgo && (name.indexOf('pgo') > 0)) {
+        return {};
+    }
+
+    return { name: name, signature: signature, platform: platform,
+             subtestSignatures: subtestSignatures };
+  };
+}]);
+
+comparetalos.controller('CompareCtrl', [ '$state', '$stateParams', '$scope', '$rootScope', '$location',
+                              '$modal', 'thServiceDomain', '$http', '$q', '$timeout', 'getSeriesSummary',
+  function CompareCtrl($state, $stateParams, $scope, $rootScope, $location, $modal,
+                    thServiceDomain, $http, $q, $timeout, getSeriesSummary) {
+
+    var availableColors = [ 'red', 'green', 'blue', 'orange', 'purple' ];
+
+    function displayComparision() {
+//TODO: why do we need so much history?
+      $scope.timeRange = 2592000; // last 30 days
+      $scope.testList = [];
+
+//TODO: how to input these variables
+$scope.e10s = false;
+$scope.pgo = false;
+
+      var signatureURL = thServiceDomain + '/api/project/' + $scope.originalProject + 
+          '/performance-data/0/get_performance_series_summary/?interval=' +
+          $scope.timeRange;
+
+      $http.get(signatureURL).then(
+        function(response) {
+          var data = response.data;
+          var seriesList = [];
+          var selectedSignatures = [];
+
+          var selectedSignaturesCount = 0;
+          Object.keys(data).forEach(function(signature) {
+            var seriesSummary = getSeriesSummary(signature,
+                                                 data[signature],
+                                                 optionCollectionMap,
+                                                 $scope.pgo,
+                                                 $scope.e10s);
+
+            if (seriesSummary == {}) {
+              return;
+            }
+
+            var testname = seriesSummary.name;
+            if ($scope.testList.indexOf(testname) === -1) {
+              $scope.testList.push(testname);
+            }
+
+            // Bug 1153301, query too long
+            if (seriesSummary.signature !== undefined && selectedSignaturesCount < 20) {
```

Comment:
As I mentioned earlier, we can batch requests to get this information and wait on all of them be complete via $q promises. See this example in perf.js: https://github.com/mozilla/treeherder-ui/blob/master/webapp/app/js/perf.js#L524